atc/db/build_test.go

```diff
@@ -1259,13 +1259,67 @@ var _ = Describe("Build", func() {
 				Expect(buildOutputs[0].Version).To(Equal(outputVersion))
 			})
 
-			Context("with a job in a separate team downstream of the same resource config", func() {
+			Context("with a job in a separate team downstream of the same non-trigger resource config", func() {
+				var otherScenario *dbtest.Scenario
+
+				var beforeTime, otherJobBeforeTime time.Time
+				var otherTeamBeforeTime, otherTeamOtherJobBeforeTime time.Time
+
+				BeforeEach(func() {
+					otherScenario = dbtest.Setup(
+						builder.WithTeam("other-team"),
+						builder.WithPipeline(pipelineConfig),
+						builder.WithResourceVersions("some-resource", atc.Version{"some": "version"}),
+						builder.WithResourceVersions("some-other-resource", atc.Version{"some": "other-version"}),
+					)
+
+					beforeTime = scenario.Job("some-job").ScheduleRequestedTime()
+					otherTeamBeforeTime = otherScenario.Job("some-job").ScheduleRequestedTime()
+
+					otherJobBeforeTime = scenario.Job("some-other-job").ScheduleRequestedTime()
+					otherTeamOtherJobBeforeTime = otherScenario.Job("some-other-job").ScheduleRequestedTime()
+				})
+
+				It("requests schedule on jobs which use the same config", func() {
+					Expect(scenario.Job("some-job").ScheduleRequestedTime()).To(BeTemporally(">", beforeTime))
```

Comment:
why this one is scheduled? Is there somewhere a pending build is created for the job?

Reply:
Yes, there is a `pending` job build for `some-job` in `scenario`.

* https://github.com/concourse/concourse/pull/8504/files#diff-f1d897e0cae526cb4f902a4aa326a976ebd9f07411cd6e7eb2896149c73168edL1210-L1215